third_party/move/move-binary-format/src/serializer.rs

```diff
@@ -496,7 +496,14 @@ fn serialize_function_handle(
     serialize_ability_sets(binary, &function_handle.type_parameters)?;
     if major_version >= VERSION_7 {
         serialize_access_specifiers(binary, &function_handle.access_specifiers)
-    } else if function_handle.access_specifiers.is_some() {
+    } else if function_handle.access_specifiers.is_some()
```

Comment:
If bytecode version is 6 or below, check whether the access_specifier is the traditional one